### PR TITLE
Optimize `parking management` solution

### PR DESCRIPTION
**Pull request by RizwanMunawar**
@glenn-jocher 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR refactors the parking management guide and code, simplifying the process and data handling.

### 📊 Key Changes
- Consolidated initialization of the parking manager with JSON and model file paths.
- Simplified code by removing unnecessary variable assignments.
- Changed bounding box handling to more generic "region data".
- Streamlined the display and processing of parking data.

### 🎯 Purpose & Impact
- **Clarity**: Simplifies setup and processing, making it easier for users to manage parking data with the YOLO model.
- **Efficiency**: Code refactoring improves readability and potentially speeds up processing with fewer redundant operations.
- **User Experience**: Easier tracking and management of parking space data with consolidated and clearer instructions. 🚗📈

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline the review process. An Ultralytics engineer will be with you shortly. Meanwhile, please review the following checklist to ensure everything is in order:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description. Link any [relevant issues](https://github.com/ultralytics/ultralytics/issues) and ensure your commit messages are clear and concise.
- ✅ **Synchronize with Source**: Make sure your PR is up-to-date with the `ultralytics/ultralytics` `main` branch. If not, update it by clicking 'Update branch' or running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify that all [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are successful. Address any issues if they arise.
- ✅ **Update Documentation**: Amend any necessary [documentation](https://docs.ultralytics.com) for the features you modified or added.
- ✅ **Add Tests**: If applicable, include or update tests to ensure your changes are covered, and check that all tests pass successfully.
- ✅ **Sign the CLA**: Please confirm that you have signed the [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first PR by stating, "I have read the CLA Document and I sign the CLA" in a new comment.
- ✅ **Minimize Changes**: Keep your changes to the **minimum** necessary for the feature improvement or bug fix. _As Bruce Lee wisely said, "It is not daily increase but daily decrease, hack away the unessential."_

If your PR involves bug reporting and does not include a Minimum Reproducible Example (MRE), kindly add one to aid the review process.

For more information, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). If you have any questions, feel free to leave a comment. Thank you for contributing to Ultralytics! 🎉🚀

File: ultralytics/solutions/parking_management.py [hunk 1]
```diff
@@ -160,102 +148,85 @@ class ParkingManagement:
...
-            model_path (str): Path to the YOLOv8 model.
-            txt_color (tuple): RGB color tuple for text.
-            bg_color (tuple): RGB color tuple for background.
-            occupied_region_color (tuple): RGB color tuple for occupied regions.
-            available_region_color (tuple): RGB color tuple for available regions.
-            margin (int): Margin for text display.
+            model (str): Path to the YOLOv8 model.
+            json_file (str): file that have all parking slot points data
+            orc (tuple): RGB color tuple for occupied regions.
+            arc (tuple): RGB color tuple for available regions.
```
Comment (glenn-jocher) [inline ultralytics/solutions/parking_management.py:163]: @RizwanMunawar I think we're going backwards here with the variable names. Variables should be human-readable and easy to understand, but names like `orc` and `arc` are not readable and do not follow Python best practices.

I would create variable names just like you see the best coding models create, i.e. if GPT-4o would not use this variable name then I would avoid it.

**Review by glenn-jocher — CHANGES_REQUESTED** [review]

File: ultralytics/solutions/parking_management.py (hunk 1 above)
Comment (RizwanMunawar) [reply ultralytics/solutions/parking_management.py:163]: @glenn-jocher Yes got it. Let me update it. Thanks!!!

**Comment by glenn-jocher** [comment]
@RizwanMunawar PR merged!

**Pull request merged**